Tests/BraveSharedTests/NSURLExtensionsTests.swift

```diff
@@ -725,25 +725,55 @@ class NSURLExtensionsTests: XCTestCase {
   }
 
   func testTypedDisplayString() {
-    let testURL1 = URL(string: "https://www.youtube.com")
-    let testURL2 = URL(string: "http://google.com")
-    let testURL3 = URL(string: "www.brave.com")
-    let testURL4 = URL(string: "http://brave.com/foo/")
-    let testURL5 = URL(string: "http://brave.com/foo")
-
     func checkDisplayURLString(testURL: URL?, displayString: String) {
       if let actual = testURL?.typedDisplayString {
         XCTAssertEqual(actual, displayString)
       } else {
         XCTFail("Actual url is nil")
       }
     }
-
-    checkDisplayURLString(testURL: testURL1, displayString: "www.youtube.com")
-    checkDisplayURLString(testURL: testURL2, displayString: "google.com")
-    checkDisplayURLString(testURL: testURL3, displayString: "www.brave.com")
-    checkDisplayURLString(testURL: testURL4, displayString: "brave.com/foo")
-    checkDisplayURLString(testURL: testURL5, displayString: "brave.com/foo")
-
+    
+    let urlMap = [
+      URL(string: "https://www.youtube.com"): "www.youtube.com",
+      URL(string: "http://google.com"): "google.com",
+      URL(string: "www.brave.com"): "www.brave.com",
+      URL(string: "http://brave.com/foo/"): "brave.com/foo",
+      URL(string: "http://brave.com/foo"): "brave.com/foo",
+      URL(string: "blob://http://brave.com/foo"): "http://brave.com/foo",
+      URL(string: "file:///Users/brave/documents/foo.txt"): "/Users/brave/documents/foo.txt",
+      URL(string: "file://http://brave.com/foo.txt"): "http://brave.com/foo.txt"
+    ]
+    
+    urlMap.forEach {
+      checkDisplayURLString(testURL: $0, displayString: $1)
+    }
+  }
+  
+  func testDisplayString() {
+    func checkDisplayURLString(testURL: URL?, displayString: String) {
+      if let actual = testURL?.displayURL?.absoluteString {
+        XCTAssertEqual(actual, displayString)
+      } else {
+        XCTFail("Actual url is nil")
+      }
+    }
+    
+    let urlMap = [
+      URL(string: "https://www.youtube.com"): "https://www.youtube.com",
+      URL(string: "http://google.com"): "http://google.com",
+      URL(string: "www.brave.com"): "www.brave.com",
+      URL(string: "http://brave.com/foo/"): "http://brave.com/foo/",
+      URL(string: "http://brave.com/foo"): "http://brave.com/foo",
+      URL(string: "blob://http://brave.com/foo"): "blob://http//brave.com/foo",
+      URL(string: "blob://02C00302-CE62-4DAE-AD70-FDEE19594856"): "blob://02C00302-CE62-4DAE-AD70-FDEE19594856",
+      URL(string: "file:///Users/brave/documents/foo.txt"): "file://foo.txt",
+      URL(string: "file://http://brave.com/foo.txt"): "file://foo.txt",
```

Comment:
Once you merge https://github.com/brave/brave-ios/pull/6250, will the `displayString` be different?

Reply:
No it will not be different. Once merged, it means the user cannot `type` file URLs into the URL bar.
But file URLs can still be displayed if the user tapped on a link to a file and it triggers a download or display of that file :)

Reply:
But do we even support `file://` schemes on iOS? `http` to  `file` links won't work for example. How would a user navigate to `file://` urls in the first place?

Reply:
They can't. So for example, when we download files for the user, we do NOT expose the actual local file link, we only expose the file itself via the `Files` app on iOS (so we don't have code for this, and it's handled by the system).

In other words, it doesn't really matter if the user types a file URL into the URL bar, it'll do nothing and it will never load. That's because all files stored in the app is stored like:
`/var/mobile/applications/UUID-v4-here/data/application support/some/path/here/file.extension`

So unless the user knows that `UUID-v4`, this is impossible. It isn't possible for the user to ever get that file path with the UUID in Brave-iOS anyway.

I fixed this bug was just because it's super annoying that the browser attempts to load `file://` URLs, and it never loads, and time-out is too long, so the loader looks stuck. Also fixed it because `hackerone` :D